zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs

```diff
@@ -382,41 +402,42 @@ mod tests {
     }
 
     proptest! {
-        #[test]
-        fn select_spendable_notes( sapling_value in 0..10_000_000u32,
-            orchard_value in 0..10_000_000u32,
-            target_value in 0..10_000_000u32,
-        ) {
-            let mut transaction_records_by_id = TransactionRecordsById::new();
-            transaction_records_by_id.insert_transaction_record(nine_note_transaction_record(
-                1_000_000_u64,
-                1_000_000_u64,
-                1_000_000_u64,
-                sapling_value as u64,
-                1_000_000_u64,
-                1_000_000_u64,
-                orchard_value as u64,
-                1_000_000_u64,
-                1_000_000_u64,
-            ));
-
-            let target_amount = NonNegativeAmount::const_from_u64(target_value as u64);
-            let anchor_height: BlockHeight = 10.into();
-            let spendable_notes =
-                zcash_client_backend::data_api::InputSource::select_spendable_notes(
-                    &transaction_records_by_id,
-                    AccountId::ZERO,
-                    target_amount,
-                    &[ShieldedProtocol::Sapling, ShieldedProtocol::Orchard],
-                    anchor_height,
-                    &[],
-                ).unwrap();
-            prop_assert_eq!(spendable_notes.sapling().len(), 1);
-            prop_assert_eq!(spendable_notes.orchard().len(), 1);
-        }
+        // TODO: rewrite select_spendable test suite to test a range of cases and target edge cases correctly
```

Comment:
I agree!  I think this is high priority..  I suppose we can land your improvements already, but we may as well add sufficient testing.